### ISSUE TITLE
Prod/Loss rate plotting tools

### ISSUE THREAD

**Issue by xcy12** [AtChem/AtChem2#426]
Hi, I wonder is there some plotting tool for total OH/HO2/RO2 production or loss rate?

**Comment by rs028** [comment]
I am afraid not. But we are happy to accept contributions if you are writing one! :)

**Comment by xcy12** [comment]
> I am afraid not. But we are happy to accept contributions if you are writing one! :)

Thank you. I'm trying to plot HOx and ROx. I will share if I succeed. It may take time because I just started to write code.

**Comment by spco** [comment]
If it helps, you should be able to modify an existing script from https://github.com/AtChem/AtChem2/tree/master/tools/plot for your purposes. The data should of course be available in `environmentVariables.output`, so it should be a nice little script if you are willing to submit it 😄

**Comment by rs028** [comment]
Those scripts already plot `environmentVariables.output`. I think that @xcy12 means `productionRates.output` and `lossRates.output`, which are a bit more complex to analyze. It would be great to have something to deal with that for sure :)

**Comment by xcy12** [comment]
> Those scripts already plot `environmentVariables.output`. I think that @xcy12 means `productionRates.output` and `lossRates.output`, which are a bit more complex to analyze. It would be great to have something to deal with that for sure :)

Yes, I mean production/loss Rate.output. It's not easy because of the huge number of reactions. Take RO2 as an example, the script first needs to recognize which one produces RO2, after then make a sum of the reaction rates for all the selected reactions. By the way, RO2 also has many kinds, which makes it harder to achieve. 
I will try and find some friends together to do this. 
Thanks a lot @spco @rs028

**Comment by rs028** [comment]
It's probably easier if you start with an individual species like HO2 or CH3O2 before trying to do RO2.

**Comment by AlfredMayhew** [comment]
Hi,
I have developed a set of tools to plot the productionRates.output and lossRates.output files produced by AtChem2. I've uploaded them to [this repo](https://github.com/AlfredMayhew/ROPAtChem2). There is currently only one script there which plots the production and loss rates of a given set of species as stackplots, but I do also have scripts for kOH and kNO3 analysis which I plan to tidy up and share at some point.
If there's any suggestions for how to make this more useful for others (I realise it's not the cleanest code) then let me know and I'll see what I have time for. I'm also happy for any part of this to be incorporated into the plotting tools in AtChem2 if that's helpful.

Cheers,
Alfie
